util/windows/utils_windows.go

```diff
@@ -0,0 +1,148 @@
+			return idtools.Identity{SID: userName}, nil
+		}
+	}
+
+	// We test for well known accounts that should exist inside any system. This has the potential
+	// to fail if the usernames/group names differ in the container, as is the case on internationalized
+	// versions where the builtin account and group names may have names in the local language.
+	// If the user specified an internationalized version of the account name, but the host is in English,
+	// this lookup will most likely fail and we will fall back to running get-account-info inside the container.
+	// This should however catch most of the cases when well known accounts/groups are used.
+	sid, _, accountType, err := syscall.LookupSID("", userName)
+	if err == nil {
+		if accountType == syscall.SidTypeAlias || accountType == syscall.SidTypeWellKnownGroup {
+			sidAsString, err := sid.String()
+			if err == nil {
+				return idtools.Identity{SID: sidAsString}, nil
+			}
+		}
+	}
+
+	// Last resort.
+	// The equivalent in Windows of /etc/passwd and /etc/group is a registry hive called SAM which can be found
+	// on any windows system in: C:\Windows\System32\config\SAM.
+	//
+	// This hive holds all user information on a particular system, including the SID of the user we care
+	// about. The bad news is that the data structures in this hive are completely undocumented and there
+	// is no API we can call to load the security info inside an offline SAM hive. We can load it as a
+	// registry hive, but parsing the data structures it holds is not documented. It's not impossible to do,
+	// but in the absence of a supported API to do this for us, we risk that sometime in the future our parser
+	// will break.
+	//
+	// That being said, we have no choice but to execute a command inside the rootMount and attempt to get the
+	// SID of the user we care about using officially supported Windows APIs. This obviously adds some overhead.
+	//
+	// TODO(gsamfira): Should we use a snapshot of the rootMount?
+	ident, err := GetUserIdentFromContainer(ctx, exec, rootMount, userName)
+	if err != nil {
+		return idtools.Identity{}, errors.Wrap(err, "getting account SID from container")
+	}
+	return ident, nil
+}
+
+func GetUserIdentFromContainer(ctx context.Context, exec executor.Executor, rootMount snapshot.Mountable, userName string) (idtools.Identity, error) {
+	var ident idtools.Identity
+	rootMounts, release, err := rootMount.Mount()
+	if err != nil {
+		return ident, errors.Wrap(err, "getting root mounts")
+	}
+	defer release()
+
+	if len(rootMounts) > 1 {
+		return ident, fmt.Errorf("unexpected number of root mounts: %d", len(rootMounts))
+	}
+
+	stdout := &bytesReadWriteCloser{
+		bw: &bytes.Buffer{},
+	}
+	stderr := &bytesReadWriteCloser{
+		bw: &bytes.Buffer{},
+	}
+
+	defer stdout.Close()
+	defer stderr.Close()
+
+	procInfo := executor.ProcessInfo{
+		Meta: executor.Meta{
+			Args: []string{"get-user-info", userName},
+			User: "ContainerAdministrator",
```

Comment:
Can this run as `ContainerUser`?

Reply:
Returns ```Access Denied``` if I remember correctly. Will double check.